packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -3399,6 +3401,24 @@ class RenderSemanticsAnnotations extends RenderProxyBox {
       markNeedsSemanticsUpdate();
   }
 
+  /// The handler for [SemanticsActio.setSelection].
+  ///
+  /// This handler is invoked when the user either wants to change the currently
+  /// selected text in a text field or change the position of the cursor.
+  ///
+  /// TalkBack users can trigger this handler by selecting "Move cursor to
+  /// beginning/end" or "Select all" from the Local Context Menu.
+  SetSelectionHandler get onSetSelection => _onSetSelection;
+  SetSelectionHandler _onSetSelection;
+  set onSetSelection(SetSelectionHandler handler) {
+    if (_onSetSelection == handler)
+      return;
+    final bool hadValue = _onSetSelection != null;
+    _onSetSelection = handler;
+    if ((handler != null) != hadValue)
```

Comment:
nit: I'd create a variable `final bool hasValue = handler != null;` so the `if` expression is more clear.

Reply:
I left this one as is for now because all the other setters for the various handlers in this file follow the same pattern.